### DISCUSSION TITLE
umi4 通过plugin.ts 配置 optimization.splitChunks 后白屏问题

### DISCUSSION THREAD

**Discussion by hexhaa** [umijs/umi#11135] [Q&A]
根目录创建 plugin.ts 文件，添加如下配置，npm run build后可以看到分包和压缩都已经生效，可是npm run dev启动的开发服务后浏览器访问白屏。请问有没有碰到相同问题的伙伴？
- 看到其他讨论提到配置 chunks，可是 umi4 的 config 中已不支持配置 chunks
- 也尝试把分包配置 splitChunks 里的 chunks 配置为 async，仍然白屏，如果配置为 initial 则分包失效
- 也尝试在 config 中配置 umi4 自带的 codeSplitting，但分包效果并不理想

`
import type { IApi } from 'umi';

const CompressionWebpackPlugin = require('compression-webpack-plugin')
const prodGzipList = ['js', 'css']

export default (api: IApi) => {
  api.chainWebpack((memo, { env, webpack }) => {
    memo
      .plugin('compression-webpack-plugin')
      .use(CompressionWebpackPlugin, [{
        algorithm: 'gzip',
        test: new RegExp('\\.(' + prodGzipList.join('|') + ')$'),
        threshold: 10240,
        minRatio: 0.3
      }])
    memo
      .optimization
      .minimize(true)
      .usedExports(true)
      .splitChunks({
        chunks: 'all',
        minSize: 100000,
        minChunks: 2,
        automaticNameDelimiter: '.',
        maxAsyncRequests: 5,
        maxInitialRequests: 3,
        cacheGroups: {
          vendors: {
            name: 'vendors',
            test: /^.*node_modules[\\/](?!lodash|monaco-editor|@wangeditor|echarts|pdfjs-dist|xlsx|luckyexcel|@fortune-sheet|antd|@ant-design|antd-mobile|jsplumb).*$/,
            chunks: 'all',
            enforce: true,
            priority: 20,
          },
          lodash: {
            name: 'lodash',
            test: /[\\/]node_modules[\\/]lodash[\\/]/,
            chunks: 'all',
            enforce: true,
            priority: 10
          },
          monacoeditor: {
            name: 'monacoeditor',
            test: /[\\/]node_modules[\\/]monaco-editor[\\/]/,
            chunks: 'all',
            enforce: true,
            priority: 10
          },
          wangeditor: {
            name: '@wangeditor',
            test: /[\\/]node_modules[\\/]@wangeditor[\\/]/,
            chunks: 'all',
            enforce: true,
            priority: 10
          },
          echarts: {
            name: 'echarts',
            test: /[\\/]node_modules[\\/]echarts[\\/]/,
            chunks: 'all',
            enforce: true,
            priority: 10
          },
          pdfjs: {
            name: 'pdfjs-dist',
            test: /[\\/]node_modules[\\/]pdfjs-dist[\\/]/,
            chunks: 'all',
            enforce: true,
            priority: 10
          },
          xlsx: {
            name: 'xlsx',
            test: /[\\/]node_modules[\\/]xlsx[\\/]/,
            chunks: 'all',
            enforce: true,
            priority: 10
          },
          luckyexcel: {
            name: 'luckyexcel',
            test: /[\\/]node_modules[\\/]luckyexcel[\\/]/,
            chunks: 'all',
            enforce: true,
            priority: 10
          },
          fortune: {
            name: 'fortune',
            test: /[\\/]node_modules[\\/]@fortune-sheet[\\/]/,
            chunks: 'all',
            enforce: true,
            priority: 10
          },
          antd: {
            name: 'antd',
            test: /[\\/]node_modules[\\/]antd[\\/]/,
            chunks: 'all',
            enforce: true,
            priority: 10
          },
          antdicon: {
            name: '@ant-design',
            test: /[\\/]node_modules[\\/]@ant-design[\\/]/,
            chunks: 'all',
            enforce: true,
            priority: 10
          },
          antdmobile: {
            name: 'antd-mobile',
            test: /[\\/]node_modules[\\/]antd-mobile[\\/]/,
            chunks: 'all',
            enforce: true,
            priority: 10
          },
          jsplumb: {
            name: 'jsplumb',
            test: /[\\/]node_modules[\\/]jsplumb[\\/]/,
            chunks: 'all',
            enforce: true,
            priority: 10
          }
        }
      })
  });
};
`

备注：在其他使用 umi3 的项目中，使用 config.ts 中的 chainWebpack 做过代码分割，可以很好实现自定义分包和压缩。

**Comment by fz6m** [comment]
在 umi 4 中默认自动按页拆包，如需进一步拆分，请调整拆分策略或手动按组件级别拆分，可参考 [拆分指南](https://umijs.org/blog/code-splitting) 。

不支持自定义配置 splitChunk 。
